go/vt/mysqlctl/builtinbackupengine_test.go

```diff
@@ -158,6 +171,93 @@ func TestExecuteBackup(t *testing.T) {
 	assert.False(t, ok)
 }
 
+// TestExecuteBackupWithCancelledContext test the ability of backup to gracefully
+// handle cases where we encounter error due to any reasons for e.g context cancel etc.
+// Process should not panic in these situations.
+func TestExecuteBackupWithCancelledContext(t *testing.T) {
+	// Set up local backup directory
+	id := fmt.Sprintf("%d", time.Now().UnixNano())
+	backupRoot := fmt.Sprintf("testdata/builtinbackup_test_%s", id)
+	filebackupstorage.FileBackupStorageRoot = backupRoot
+	require.NoError(t, createBackupDir(backupRoot, "innodb", "log", "datadir"))
+	dataDir := path.Join(backupRoot, "datadir")
+	// Add some files under data directory to force backup to execute semaphore acquire inside
+	// backupFiles() method (https://github.com/vitessio/vitess/blob/main/go/vt/mysqlctl/builtinbackupengine.go#L483).
+	require.NoError(t, createBackupDir(dataDir, "test1"))
+	require.NoError(t, createBackupDir(dataDir, "test2"))
+	require.NoError(t, createBackupFiles(path.Join(dataDir, "test1"), 2, "ibd"))
+	require.NoError(t, createBackupFiles(path.Join(dataDir, "test2"), 2, "ibd"))
+	defer os.RemoveAll(backupRoot)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Why is this sleep required?

Reply:
yeah I tried to simulate that context cancelled after sometime but here its not need since code serialized anyways.